### PR TITLE
Implementing ++ and -- operator and writing Unit Test

### PR DESCRIPTION
**Pull request by AshutoshGuptaPhilips**

### CODE REVIEW

**Review by GaneshDPhilips — COMMENTED** [review]
Is this file required.. This is generated on run time right

**Review by GaneshDPhilips — COMMENTED** [review]
Can we make Increment and decrement as two different operation type and two seperate files?

File: SRC/ReplaceIncrement.cpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+bool ReplaceIncrement::Mutate(std::string& line)
+{
+	const char* c = line.c_str();
+	bool result = false;
+
+	cout << "\n ReplaceIncrement : " << line << "\n";
+	if (line.find("::hl7") != string::npos)
+	{
+		return  result;
+	}
```
Comment (SwatiNPhilips) [inline SRC/ReplaceIncrement.cpp]: Remove HL7 related calls and make code generic

File: SRC/ReplaceIncrement.cpp [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	std::string::const_iterator it = line.begin();
+	if ((line.find(INCREMENT[0].c_str()) != string::npos))
+	{
+		line.replace(line.find(INCREMENT[0].c_str()), SIZEOFINCREMENT, DECREMENT[0].c_str());
+		result = true;
+	}
+    else if ((line.find(INCREMENT[1].c_str()) != string::npos))
```
Comment (SwatiNPhilips) [inline SRC/ReplaceIncrement.cpp]: If there is a function that has ++ and next line it has += , is both lines getting mutated ?

File: SRC/ReplaceMultiplicationWithMinus.cpp [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+}
+
+bool ReplaceMultiplicationWithMinus::Mutate(std::string& line)
+{
+	bool result = false;
+	cout << "\n ReplaceMultiplicationWithMinus : " << line << "\n";
+    for(unsigned int i = 0; i < line.length(); i++)
+    {
+        if(line[i] == MULTIPLICATION && i < line.length() - 1)
```
Comment (SwatiNPhilips) [inline SRC/ReplaceMultiplicationWithMinus.cpp]: Does this exclude the pointer dereference ?

File: SRC/ReplaceMultiplicationWithMinus.cpp [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+static constexpr char PLUS = '+';
+static constexpr char MINUS = '-';
+static constexpr char MULTIPLICATION = '*';
+static constexpr char DIVISION = '/';
+static constexpr char GREATER = '>';
+static constexpr char DOT = '.';
+static constexpr char WHITESPACE = ' ';
+static constexpr char SEMICOLON = ';';
+
+ReplaceMultiplicationWithMinus::ReplaceMultiplicationWithMinus()
```
Comment (SwatiNPhilips) [inline SRC/ReplaceMultiplicationWithMinus.cpp]: We can either replace multiplication with + or - why do we have both the classes ?

File: SRC/ReplaceDecrement.cpp [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+{
+
+}
+
+ReplaceDecrement::~ReplaceDecrement()
+{
+
+}
+
+bool ReplaceDecrement::Mutate(std::string& line)
```
Comment (SwatiNPhilips) [inline SRC/ReplaceDecrement.cpp]: fix indentation

File: SRC/ReplaceDivisionWithMinus.cpp [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+}
+
+bool ReplaceDivisionWithMinus::Mutate(std::string& line)
+{
+	bool result = false;
+	cout << "\n ReplaceDivisionWithMinus : " << line << "\n";
+    for(unsigned int i = 0; i < line.length(); i++)
+    {
+        if(line[i] == DIVISION)
```
Comment (SwatiNPhilips) [inline SRC/ReplaceDivisionWithMinus.cpp]: Do we consider abs path which also has "/" ?

File: SRC/ReplaceDivisionWithMinus.cpp [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+ReplaceDivisionWithMinus::~ReplaceDivisionWithMinus()
+{
+
+}
+
+bool ReplaceDivisionWithMinus::Mutate(std::string& line)
+{
+	bool result = false;
+	cout << "\n ReplaceDivisionWithMinus : " << line << "\n";
+    for(unsigned int i = 0; i < line.length(); i++)
```
Comment (SwatiNPhilips) [inline SRC/ReplaceDivisionWithMinus.cpp]: Why do we iterate over each char ? Cant we use find here ?

**Review by SwatiNPhilips — COMMENTED** [review]
General review : 
We should have only 1 mutate for each operator, i.e ,
Multiplication replaced with +
Division replaced with -
++, += with --, -=
--, -= with ++, +=

**Pull request closed without merging**